File: api-gateway/src/main/java/org/lite/gateway/entity/ApiMetric.java
package org.lite.gateway.entity;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import java.time.LocalDateTime;

@Data
@Document("apiMetrics")
public class ApiMetric {
    @Id
    private String id;
    private String fromService; //originated microservice
    private String toService; //target microservice
    private LocalDateTime timestamp;
    private long duration; // in milliseconds
    private String routeIdentifier;  // unique route identifier from ApiRoute
    private String interactionType;  // APP_TO_APP or USER_TO_APP
    private String gatewayBaseUrl;   // The base URL of the gateway
    private String pathEndPoint;     // Endpoint path after the base URL
    private String queryParameters;  // Parameters for GET requests
    private String requestPayload;   // Body data for POST requests
    private boolean success;         // false if exception is thrown
}

File: edge-service/backend/models/ApiMetric.js
import mongoose from 'mongoose';

const apiMetricSchema = new mongoose.Schema({
    fromService: { type: String, required: true },
    toService: { type: String, required: true },
    timestamp: { type: Date, required: true },
    duration: { type: Number, required: true },
    routeIdentifier: { type: String, required: true },
    interactionType: { type: String, required: true },
    gatewayBaseUrl: { type: String, required: true },
    pathEndPoint: { type: String, required: true },
    queryParameters: { type: String, default: '' },
    success: { type: Boolean, required: true },
    _class: { type: String, required: true },
}, { collection: 'apiMetrics' });

export default mongoose.model('ApiMetric', apiMetricSchema);
